alacritty_terminal/src/grid/mod.rs

```diff
@@ -458,7 +455,7 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
                     row = Row::from_vec(wrapped, occ);
 
                     // Increase scrollback history
-                    self.scroll_limit = min(self.scroll_limit + 1, self.max_scroll_limit);
+                    history_size = min(history_size + 1, self.max_scroll_limit);
```

Comment:
It seems like you're just modifying a variable that is never used? Since the `self.history_size()` returns a copy and not a mutable reference.

So this is buggy.

Reply:
I know that's why I asked that I don't really know hot to update this thing in a more better way, since you're speculating on a storage, and self.history() is also speculating on a history, so you have 2 mut refs that can update the storage/invalidate interator(not in rust). Also you're updating the entire storage in the end anyway, so I thought that this scroll_limit thing was just useful only in a context of this `for loop`, since I don't see how to update it after the storage, and I don't see any sign, that this introduce a bug.

Reply:
I'd have to test things myself, but maybe it works to increase/decrease the scroll limit after all lines are processed.

Alternatively it might need some slight tweaks to the logic, but it shouldn't be that big of a problem since we're creating a new buffer anyways. So we have good control over the raw storage here.